### ISSUE TITLE
Build errors on latest Xcode 10 - Xcode is looking for macOS headers, but it's an iOS app

### ISSUE THREAD

**Issue by koenvanderdrift** [core-plot/core-plot#392]
Recently, I'm getting the following build errors when building my iOS app with CorePlot using Xcode 10 beta. 

```
/FrameWorks/core-plot/framework/MacOnly/CPTPlatformSpecificDefines.h:3:9: Unknown type name 'NSImage'
/FrameWorks/core-plot/framework/MacOnly/CPTPlatformSpecificDefines.h:4:9: Unknown type name 'NSEvent'; did you mean 'UIEvent'?
/FrameWorks/core-plot/framework/MacOnly/CPTGraphHostingView.h:1:9: 'Cocoa/Cocoa.h' file not found
```

Which is strange, since it is an iOS app, not a macOS app and the target is linked to `libCorePlot-CocoaTouch.a` 

I don't see see how this could have happened - I'm sure there is a simple explanation. 

Any ideas what could have happened and how to fix it?

**Comment by eskroch** [comment]
I suspect this is related to issue #386. I haven't found a solution yet.

**Comment by eskroch** [comment]
Check your header search paths. I've seen problems with finding headers when building a Core Plot framework as a dependent project, but not with the iOS static library.

**Comment by koenvanderdrift** [comment]
Right now it is `../../FrameWorks/core-plot/framework` - recursive, do I need to change that?

**Comment by msimms** [comment]
I had the same issue. The problem is that if you have recursive search enabled for your header files, XCode will search the `MacOnly` subdirectory before searching the `IPhoneOnly` subdirectory. Of course, after you turn recursive search off, you'll need to add the subdirectories you actually need back into your search path, namely `core-plot/framework`, `core-plot/framework/Source`, `and core-plot/framework/IPhoneOnly`.

**Comment by laviniuc** [comment]
works also if i removed the maconly folder completely

**Comment by wspires** [comment]
> works also if i removed the maconly folder completely (although it's probably too radical)

Thanks, I'm running into the same issue and removing framework/MacOnly was the only way I could get a successful build. CorePlot has macro guards for conditional compiling components on iOS versus macOS, but perhaps something changed with Xcode 10/iOS 12?

**Comment by JasonMarzBingBang** [comment]
Same issue here.

**Comment by hwiesmann** [comment]
I solved the problem when by disabling the analyzer in CorePlot.xconfig.

**Comment by dasboos** [comment]
I am using Xcode 10.1 and I was able to resolve the errors above (and many more) simply by going to the Project Navigator -> Frameworks and selecting my CorePlot.xcodeproj.  Then in the project settings window that appeared, I first selected Project -> CorePlot in the upper left main window, then I went to the "Build Settings" tab and Expanded the 'Architectures' item and found the  item "Base SDK".  That item was currently set to 'MacOS' and I changed it to iOS, then cleaned, closed xcode, resarted xcode, and rebuilt.
